--- m4/boost.m4	2019-09-17 13:50:36.785616947 +0200
+++ m4/boost.m4	2019-09-17 13:52:48.528367547 +0200
@@ -67,7 +67,7 @@
 dnl boost_cv_lib_version='1_37\r' for instance, which breaks
 dnl everything else.
 dnl Cannot use 'dnl' after [$4] because a trailing dnl may break AC_CACHE_CHECK
-(eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD |
+(eval "$ac_cpp -P conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD |
   tr -d '\r' |
   $SED -n -e "$1" >conftest.i 2>&1],
   [$3],
